### PR TITLE
fix(DataGrid): UXD-841 fixed ui issues related to borders

### PR DESCRIPTION
**Pull request by allison-c**
### Purpose 🚀
_clear, one sentence description of primary purpose of this PR_

### Notes ✏️

- [x] 1) removed the line between checkbox and empty column header
- [x] 1) prevent scroll bar showing when zoom in
- [x] 2) added the missing border bottom on sticky column header
- [x] 3) changed background color to white
- [x] 4) added the 2px border on both header and footer

After:
image

### Updates 📦
- [ ] MAJOR (breaking) change to _these packages_ (if the package you are breaking is a dependency in other packages, be sure to add a commit with a "BREAKING CHANGE" message to those other packages as well)
- [ ] MINOR (backward compatible) change to _these packages_
- [ ] PATCH (bug fix) change to _these packages_

### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-841-data-grid-ui-issues

### Screenshots 📸
_optional but highly recommended_

### References 🔗
_relevant Jira ticket / GitHub issues_

### CODE REVIEW

File: packages/DataGrid/src/DataGrid.Rows.js [hunk 1]
```diff
@@ -142,7 +143,11 @@ export const StickyHeaderRow = React.memo(({ data: RowData, columnIndex, style }
   return (
     <CellHeader
       role="columnheader"
-      style={{ ...style, ...styleProps }}
+      style={{
+        boxShadow: `inset 0px -1px 0 0 ${tokens.border.color}, 0px 0px 0px 1px ${tokens.border.color}`,
```
Comment (allison-c) [inline packages/DataGrid/src/DataGrid.Rows.js:147]: 2)

File: packages/DataGrid/src/DataGrid.styles.js [hunk 2]
```diff
@@ -64,6 +64,11 @@ export const Grid = styled.div.attrs(({ $width }) => {
         overflow: auto !important;
       }
 
+      .${gridId}-sticky-header {
+        box-shadow: 2px 0px 0px 0px ${tokens.border.color};
+        z-index: 1;
```
Comment (allison-c) [inline packages/DataGrid/src/DataGrid.styles.js:69]: 4)

File: packages/DataGrid/src/DataGrid.styles.js [hunk 3]
```diff
@@ -77,7 +82,7 @@ export const Grid = styled.div.attrs(({ $width }) => {
       .${gridId}-sticky-columns {
         /* The arrow navigation is sensitive with borders. If a border is used, it will cause issues keyboard navigation. */
         -ms-overflow-style: none;
-        box-shadow: 3px 0px 0px 0px ${tokens.border.color};
+        box-shadow: 2px 0px 0px 0px ${tokens.border.color};
```
Comment (allison-c) [inline packages/DataGrid/src/DataGrid.styles.js:85]: 4)

File: packages/DataGrid/src/DataGrid.styles.js [hunk 4]
```diff
@@ -217,9 +222,9 @@ export const FillerTopRight = styled.div`
 
 export const FillerBottomLeft = styled.div`
   /* this is a small square filler on the bottom left corner of the DataGrid */
-  background: ${tokens.table.header.backgroundColor};
+  background: ${tokens.color.white};
```
Comment (allison-c) [inline packages/DataGrid/src/DataGrid.styles.js:225]: 3)

File: packages/DataGrid/src/DataGrid.styles.js [hunk 5]
```diff
@@ -217,9 +222,9 @@ export const FillerTopRight = styled.div`
...
   bottom: 0;
-  box-shadow: 0 0 0 1px ${tokens.border.color};
+  box-shadow: 1px 0 0 1px ${tokens.border.color};
```
Comment (allison-c) [inline packages/DataGrid/src/DataGrid.styles.js:227]: 4)

File: packages/DataGrid/src/components/columns/renderColumnExpand.js [hunk 6]
```diff
@@ -61,6 +62,11 @@ export default function renderColumnExpand(options = {}) {
     <ColumnDefinition
       header={() => null}
       headerA11yText={() => "Expand Row"}
+      headerProps={() => ({
+        style: {
+          boxShadow: `inset 0px -1px 0 0 ${tokens.border.color}`,
```
Comment (allison-c) [inline packages/DataGrid/src/components/columns/renderColumnExpand.js:67]: 1)

**Review by nahumzs — APPROVED** [review]

**Pull request merged**